src/lib/pmd/PmdEngine.ts

```diff
@@ -25,20 +26,14 @@ interface PmdViolation extends Element {
 	};
 }
 
-export class PmdEngine implements RuleEngine {
-	private static THIS_ENGINE = ENGINE.PMD;
-	public static ENGINE_NAME = PmdEngine.THIS_ENGINE.valueOf();
+abstract class BasePmdEngine implements RuleEngine {
```

Comment:
Brokedown the common portion into a base class. `PmdEngine` and `CustomPmdEngine` extend this class to change their own portions.